crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use ruma::{MilliSecondsSinceUnixEpoch, OwnedUserId};
+use serde::{Deserialize, Serialize};
+use vodozemac::Ed25519PublicKey;
+
+use crate::types::DeviceKeys;
+
+/// Information on the device and user that sent the megolm session data to us
+///
+/// Sessions start off in `UnknownDevice` state, and progress into `DeviceInfo`
+/// state when we get the device info. Finally, if we can look up the sender
+/// using the device info, the session can be moved into `SenderKnown` state.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
```

Comment:
If we sorted the derives we usually  sort these by groups, first come the traits from the standard library, then the rest. Each group is alphabetically sorted.

Not really important, and `cargo fmt` can't sort these for you so I don't want to enforce a style either, but perhaps you'd be interested in following an existing convention that is used at least in some places.

Reply:
Definitely. Fixed in a39f7a5e85